tools/pubsys/src/aws/ami/register.rs

```diff
@@ -74,17 +72,17 @@ async fn _register_image(
     }
 
     // Prepare parameters for AMI registration request
-    let root_bdm = BlockDeviceMapping {
-        device_name: Some(ROOT_DEVICE_NAME.to_string()),
-        ebs: Some(EbsBlockDevice {
-            delete_on_termination: Some(true),
-            snapshot_id: Some(root_snapshot.clone()),
-            volume_type: Some(VOLUME_TYPE.to_string()),
-            volume_size: ami_args.root_volume_size,
-            ..Default::default()
-        }),
-        ..Default::default()
-    };
+    let root_bdm = BlockDeviceMapping::builder()
+        .set_device_name(Some(ROOT_DEVICE_NAME.to_string()))
+        .set_ebs(Some(
+            EbsBlockDevice::builder()
+                .set_delete_on_termination(Some(true))
+                .set_snapshot_id(Some(root_snapshot.clone()))
+                .set_volume_type(Some(VolumeType::from(VOLUME_TYPE)))
+                .set_volume_size(ami_args.root_volume_size)
+                .build(),
```

Comment:
I think these can be changed to -
```suggestion
        .device_name(ROOT_DEVICE_NAME.to_string())
        .ebs(EbsBlockDevice::builder()
                .delete_on_termination(true)
                .snapshot_id(root_snapshot.clone())
                .volume_type(VolumeType::from(VOLUME_TYPE))
                .set_volume_size(ami_args.root_volume_size)
                .build(),
```

Reply:
This definitely works, but my understanding was the setters are more idiomatic for the SDK.

Reply:
Where did you see the perspective on idiomatic use? Not disagreeing, and it's not a blocker here in any case, but for me the code that @ecpullen wrote seems preferable since it's fewer characters and we don't have to wrap everything in an `Option`.

Reply:
It was just an impression we got from the SDK team that the non-setter way might eventually be deprecated.